spec/filters/dissect_spec.rb

```diff
@@ -3,6 +3,31 @@
 require "logstash/filters/dissect"
 
 describe LogStash::Filters::Dissect do
+  class LoggerMock
```

Comment:
I think the appropriate Log4j2 strategy most similar to this is to implement something like a `TestLog4j2Appender` which appends to an array structure. This would be analogous to the `logger.subscribe(Array.new)`. So, setup configuration might have to change a little, but the assertions would be the same.

hope that makes sense. `TestLog4j2Appender` doesn't exist yet, but I will add it so that testing would be straightforward and similar to what @jordansissel recommended